src/gt4py/next/iterator/ir_utils/domain_utils.py

```diff
@@ -123,16 +126,19 @@ def translate(
                 ]
                 horizontal_sizes: dict[str, itir.Expr]
                 if symbolic_domain_sizes is not None:
-                    horizontal_sizes = {k: im.ref(v) for k, v in symbolic_domain_sizes.items()}
+                    horizontal_sizes = {
```

Comment:
Strictly unrelated to the actual PR. This just allows using literals as the symbolic domain sizes. Needed for a hack in icon4py. Didn't bother creating a PR.